packages/chevrotain/src/parse/parser/traits/recognizer_engine.ts

```diff
@@ -666,25 +662,26 @@ export class RecognizerEngine {
     }
   }
 
-  subruleInternal<T>(
+  subruleInternal<ARGS extends unknown[], R>(
     this: MixedInParser,
-    ruleToCall: (idx: number) => T,
+    ruleToCall: ParserMethod<ARGS, R>,
     idx: number,
-    options?: SubruleMethodOpts
-  ) {
+    options?: SubruleMethodOpts<ARGS>
+  ): R {
     let ruleResult
     try {
       const args = options !== undefined ? options.ARGS : undefined
-      ruleResult = ruleToCall.call(this, idx, args)
+      this.subruleIdx = idx
+      ruleResult = ruleToCall.apply(this, args)
```

Comment:
I wonder if `call` vs `apply` would have different performance here

Reply:
In invokeWithTry we are using the spread operator and then calling apply again.
So it seems we are "packing"  and "unpacking" the ARGS too many times.
- `invokeRuleWithTry(this: MixedInParser, ...args: ARGS[])` 

If we remove the spread operator there, we could then keep this line with the possibly faster `call` and avoid un-needed packs/unpacks ?

Reply:
Yes, but then the result of RULE wouldn't have the same signature as the implementation definition, which is the most intuitive. As there are going to be no arguments in most cases, I think a minor performance import would be worth it.

Reply:
I will investigate the performance impact (if any) after we merge.